pkg/operators/ustack/ustack.go

```diff
@@ -0,0 +1,275 @@
+
+			targetName, err := annotations.GetTargetNameFromAnnotation(logger, "ustack", in, userStackTargetNameAnnotation)
+			if err != nil {
+				logger.Warnf("getting target name for ustack field %q: %v", in.Name(), err)
+				continue
+			}
+			out, err := ds.AddField(targetName, api.Kind_String, datasource.WithSameParentAs(in))
+			if err != nil {
+				return err
+			}
+			converter := func(ds datasource.DataSource, data datasource.Data) error {
+				inode, _ := inodeField[0].Uint64(data)
+				// If user stacks are disabled
+				if inode == 0 {
+					return nil
+				}
+
+				stackId, _ := stackField[0].Uint32(data)
+				pidLevel0, _ := pidLevel0Field[0].Uint32(data)
+				pidnsLevel0, _ := pidnsLevel0Field[0].Uint32(data)
+				pidLevel1, _ := pidLevel1Field[0].Uint32(data)
+				pidnsLevel1, _ := pidnsLevel1Field[0].Uint32(data)
+				if pidLevel0 == 0 || pidnsLevel0 == 0 || pidLevel1 == 0 || pidnsLevel1 == 0 {
+					logger.Warn("invalid pid or pidns")
+					return nil
+				}
+				pidNumbers := []symbolizer.PidNumbers{
+					{
+						Pid:     pidLevel0,
+						PidNsId: pidnsLevel0,
+					},
+					{
+						Pid:     pidLevel1,
+						PidNsId: pidnsLevel1,
+					},
+				}
+				containerPid := uint32(0)
+				if containerPidField != nil {
+					containerPid, _ = containerPidField.Uint32(data)
+				}
+				containerName := ""
+				if containerNameField != nil {
+					containerName, _ = containerNameField.String(data)
+				}
+				comm := ""
+				if commField != nil {
+					comm, _ = commField.String(data)
+				}
+				mtimeSec, _ := mtimeSecField[0].Uint64(data)
+				mtimeNsec, _ := mtimeNsecField[0].Uint32(data)
+
+				stack := [ebpftypes.PerfMaxStackDepth]uint64{}
+				err := o.userStackMap.Lookup(stackId, &stack)
+				if err != nil {
+					logger.Warnf("stack with ID %d is lost: %s", stackId, err.Error())
+					return nil
+				}
+
+				addrs := make([]uint64, 0, len(stack))
+				for _, addr := range stack {
+					if addr == 0 {
+						break
+					}
+					addrs = append(addrs, addr)
+				}
+				task := symbolizer.Task{
+					Name:         fmt.Sprintf("%s/%s", containerName, comm),
+					PidNumbers:   pidNumbers,
+					ContainerPid: containerPid,
+					Ino:          inode,
+					MtimeSec:     int64(mtimeSec),
+					MtimeNsec:    mtimeNsec,
+				}
+				symbols, err := o.symbolizer.Resolve(task, addrs)
+				if err != nil {
+					logger.Warnf("symbolizer: %s", err)
+					return nil
+				}
+
+				outString := ""
+				for i, symbol := range symbols {
+					outString += fmt.Sprintf("[%d]%s; ", i, symbol)
+				}
+				out.Set(data, []byte(outString))
+				return nil
+			}
+			o.subscriptions[ds] = append(o.subscriptions[ds], converter)
+		}
+	}
+	return nil
+}
+
+func (o *OperatorInstance) Name() string {
+	return Name
+}
+
+func (o *OperatorInstance) Stop(gadgetCtx operators.GadgetContext) error {
+	o.symbolizer.Close()
+	return nil
+}
+
+func (o *OperatorInstance) Start(gadgetCtx operators.GadgetContext) error {
+	if len(o.subscriptions) == 0 {
+		return nil
+	}
```

Comment:
I think you can move this check to init() and return a nil instance. IG interprets nil as ignoring the operator instance, so it won't be considered.

Reply:
Do you mean `InstantiateDataOperator()` returning nil? I would have to change the `OperatorInstance.init()` prototype to return the `OperatorInstance` too...

Reply:
That was my initial idea. Another option is to it on InstantiateDataOperator just after calling init(). The whole idea is to avoid returning an instance what won't be used later on.